src/main/java/org/apache/sling/caconfig/resource/spi/ConfigurationBindingsResourceDetectionStrategy.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.caconfig.resource.spi;
+
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ConsumerType;
+
+import javax.script.Bindings;
+
+@ConsumerType
+public interface ConfigurationBindingsResourceDetectionStrategy {
+    /**
+     * Detects the resource by looking at the available bindings.
+     * @param bindings Bindings object to use for detecting the correct Resource.
+     * @return Detected Resource or {@code null} if this strategy did not detect an applicable resource.
+     */
+    @Nullable Resource detectResource(@NotNull Bindings bindings);
```

Comment:
i would prefer not using the Bindings class as input parameter here - maybe better SlingHttpServletRequest or even better the current resource of the current request. it might be that we want to use this interface as well for [SLING-7256|https://issues.apache.org/jira/browse/SLING-7256], the use case is a bit similar when injecting the fitting configuration in a sling model). so Resource might fit best?

Reply:
Hi @stefanseifert , in our case we're trying to solve another issue here as well: In an AEM context, it could be that the resource on the current request (or request.getResource(), if that's passed in this method), points to a node in /conf. 

We're using the Bindings to obtain the current page, so that it will always point to /content (and never to /conf). 

See also the other PR in the wcm.io extensions project. 

Any suggestion on how to tackle this problem?

Reply:
then we should use SlingHttpServletRequest  as parameter.
for an AEM specific strategy implementation it would be no problem getting the sling bindings from the request and reading the current page from it.
it's more flexible than directly passing in the Bindings object.

Reply:
switched to SlingHttpServletRequest as we need this interface also for SLING-7256